client.go

```diff
@@ -637,23 +663,29 @@ func (c *Client) runRepliesConsumer(inChan *amqp.Channel) error {
 		c.replyToQueueDeclareArgs,
 	)
 	if err != nil {
-		return err
+		return "", err
 	}
 
+	tag := uuid.NewString()
+
 	messages, err := inChan.Consume(
 		queue.Name,
-		"",    // consumer tag. Auto-generated by the server.
+		tag,   // consumer tag.
 		true,  // auto-ack. We don't support manual ack for the reply-to queue.
 		true,  // exclusive. We must be the only consumer.
 		false, // no-local.
 		false, // no-wait.
 		c.replyToConsumerArgs,
 	)
 	if err != nil {
-		return err
+		return "", err
 	}
 
+	wg.Add(1)
+
 	go func() {
+		wg.Done()
```

Comment:
Should this be deferred? Otherwise we just decrement immediately when we start the go routine.

```suggestion
		defer wg.Done()
```

Reply:
Thanks! Fixed!